### PR TITLE
SWARM-328: Adds Hibernate Search fraction

### PR DESCRIPTION
**Pull request by gastaldi**
Motivation:
Hibernate Search is already added as part of the Wildfly core modules. Fixes SWARM-328

Modifications:
Adds a Hibernate Search fraction that allows users to use Hibernate Search in their projects.

### CODE REVIEW

File: hibernate-search/modules/pom.xml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  ~
+  ~ Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.wildfly.swarm</groupId>
+    <artifactId>hibernate-search-parent</artifactId>
+    <version>1.0.0.Beta5-SNAPSHOT</version>
```
Comment (heiko-braun) [inline hibernate-search/modules/pom.xml]: Is the specific version here needed?

Comment (gastaldi) [inline hibernate-search/modules/pom.xml]: Yes, it belongs to the parent.

File: hibernate-search/modules/pom.xml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      <exclusions>
+        <exclusion>
+          <groupId>*</groupId>
+          <artifactId>*</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.wildfly</groupId>
+      <artifactId>wildfly-feature-pack</artifactId>
```
Comment (heiko-braun) [inline hibernate-search/modules/pom.xml]: Are the core feature packs really needed? Looking at the module descriptors it doesn't seem to be the case, but I may be wrong.

File: hibernate-search/api/src/main/java/org/wildfly/swarm/hibernate/search/HibernateSearchFraction.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * limitations under the License.
+ */
+package org.wildfly.swarm.hibernate.search;
+
+import org.wildfly.swarm.spi.api.Fraction;
+
+/**
+ * @author George Gastaldi
+ */
+public class HibernateSearchFraction implements Fraction {
```
Comment (heiko-braun) [inline hibernate-search/api/src/main/java/org/wildfly/swarm/hibernate/search/HibernateSearchFraction.java]: no configuration options in the fraction at all?

File: hibernate-search/pom.xml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  ~ Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.wildfly.swarm</groupId>
+    <artifactId>core</artifactId>
+    <version>1.0.0.Beta5-SNAPSHOT</version>
```
Comment (heiko-braun) [inline hibernate-search/pom.xml:14]: Is the version needed here?

File: hibernate-search/modules/pom.xml (hunk 2 above)
Comment (gastaldi) [inline hibernate-search/modules/pom.xml]: Good catch. Probably not, let me retest again.

File: hibernate-search/runtime/pom.xml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  ~
+  ~ Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.wildfly.swarm</groupId>
+    <artifactId>hibernate-search-parent</artifactId>
+    <version>1.0.0.Beta5-SNAPSHOT</version>
```
Comment (heiko-braun) [inline hibernate-search/runtime/pom.xml]: Same as above

File: hibernate-search/test/src/test/java/org/wildfly/swarm/hibernate/search/HibernateSearchArquillianTest.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    @Override
+    public Container newContainer(String... args) throws Exception {
+        return new Container().fraction(HibernateSearchFraction.createDefaultFraction());
+    }
+
+    @Test
+    @RunAsClient
+    public void testNothing() {
```
Comment (heiko-braun) [inline hibernate-search/test/src/test/java/org/wildfly/swarm/hibernate/search/HibernateSearchArquillianTest.java]: Can you think of actual test cases?

File: hibernate-search/pom.xml (hunk 4 above)
Comment (gastaldi) [inline hibernate-search/pom.xml:14]: Yes, it belongs to the parent.

File: hibernate-search/runtime/pom.xml (hunk 5 above)
Comment (gastaldi) [inline hibernate-search/runtime/pom.xml]: Yes, it belongs to the parent.

File: hibernate-search/api/src/main/java/org/wildfly/swarm/hibernate/search/HibernateSearchFraction.java (hunk 3 above)
Comment (gastaldi) [inline hibernate-search/api/src/main/java/org/wildfly/swarm/hibernate/search/HibernateSearchFraction.java]: I couldn't find any configuration options for this fraction (it doesn't add a subsystem to standalone.xml)

File: hibernate-search/README.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# WildFly Swarm - Hibernate Search
```
Comment (kenfinnigan) [inline hibernate-search/README.md]: Not sure we need this file. It's not a precendent we've set, and not sure the value it adds as we wouldn't expect users to be trawling through -core source for this information

**Comment by kenfinnigan** [comment]
Is there a reason that the fraction isn't implemented in the same manner as `hibernate-validator`? As far as I can see, the fraction is simply pulling in some hibernate search modules. There is no WF extension to be started and no particular handling needed by WildFly Swarm

**Comment by gastaldi** [comment]
@kenfinnigan I followed the instructions in https://wildfly-swarm.gitbooks.io/wildfly-swarm-users-guide/content/fraction_authoring.html. I guess you're right, let me change that

**Comment by kenfinnigan** [comment]
need to update the docs, as there are definitely 2 kinds of fractions that can be developed

**Comment by bobmcwhirter** [comment]
Sorry, that chapter is absolutely sub-par thus far.

File: pom.xml [hunk 8]
```diff
@@ -489,6 +489,27 @@
 
       <dependency>
         <groupId>org.wildfly.swarm</groupId>
+        <artifactId>hibernate-search</artifactId>
+        <version>1.0.0.Beta5-SNAPSHOT</version>
+      </dependency>
```
Comment (kenfinnigan) [inline pom.xml:494]: Shouldn't need anything but `hibernate-search` here now

**Comment by gastaldi** [comment]
Do not merge this yet. Doing some tests

File: pom.xml (hunk 8 above)
Comment (gastaldi) [inline pom.xml:494]: Done. Thanks

**Comment by gastaldi** [comment]
Please merge https://github.com/wildfly-swarm/wildfly-swarm/pull/3 after this PR is merged

**Pull request merged**